### DISCUSSION TITLE
About imgui and cmake

### DISCUSSION THREAD

**Discussion by zw-ray** [ocornut/imgui#5073] [New Users Build/Link/Run issues ONLY!]
I am a newer to opengl, and I really like imgui since my first sight on it.

It came to my mind that making an engine is pretty nice. But when I start using imgui, it was confused to me that such a good c++ library but without cmake building script? Sorry for that I am a newer to cmake as well.

**Comment by PathogenDavid** [comment]
In my opinion there isn't significant value in Dear ImGui providing its own CMake configuration. It's not a complicated library to build, we're better off providing build-system-agnostic documentation and letting people use what they like.

If you need an example of using CMake with Dear ImGui, there are [numerous examples](https://github.com/ocornut/imgui/pulls?q=is%3Apr+cmake+in%3Atitle). In particular see https://github.com/ocornut/imgui/pull/255.

**Comment by rokups** [comment]
Multiple build scripts are listed here: https://github.com/ocornut/imgui/wiki#building--packaging-cruft
